### ISSUE TITLE
Compilation fails on Apple M1 (ANTLR)

### ISSUE THREAD

**Issue by mpreiner** [CVC4/CVC4#5769]
Issue originally submitted by Stephen Siegel on cvc-users list.

```
I ran into this problem trying to build CVC4 1.8 from source on macOS 11.1 (Apple M1)…

[ 93%] Building CXX object src/parser/CMakeFiles/cvc4parser.dir/antlr_line_buffered_input.cpp.o
/opt/sw/CVC4-1.8/src/parser/antlr_line_buffered_input.cpp:291:10: error: cast from
      pointer to smaller type 'ANTLR3_MARKER' (aka 'int') loses information
  while ((ANTLR3_MARKER)(input->nextChar) != seekPoint) {
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [src/parser/CMakeFiles/cvc4parser.dir/antlr_line_buffered_input.cpp.o] Error 1

I figured out the problem.   The script get-antlr-3.4 attempts to guess whether the architecture is 32 or 64 bit, but for arm64 it guesses incorrectly (if MACHINE_TYPE is not 'x86_64' it concludes it is 32 bit).  The solution is to read the output from this script:

 ============== WARNING ====================
The script guessed that this machine is 32 bit.
If antlr should be built as 64 bit (i.e. -m64),
please rerun the script as
MACHINE_TYPE="x86_64" ./get-antlr3.4

[note: it should be get-antlr-3.4].   After doing what it said, the build completed and all tests passed.   It might be a good idea to update this script as arm64 becomes more widely used.

```

**Comment by mpreiner** [comment]
We decided to add explicit machine type checks for ARM/Apple M1. If a machine type could not be recognized we should fail instead of printing a warning.

**Comment by nafur** [comment]
This issue should be resolved by #6033.
